MozillaReality/FirefoxReality
# Closes #2267 AC WebExtensions + WebCompat + FxAWebChannels

Closes #2267 Closes #3339 This PR adds support for the following:
- AC 44.0.0
- AC Web Extensions bridge. We are using a `ComponentsAdapter` as a bridge between our `SessionStore` and AC's `BrowserStore` which is required by the AC WebExtensions/Features lifecycle.
- Moves our current builtin extensions (YouTube and Vimeo) to AC `WebExtensionsRuntime`.
- Adds support for the `WebCompat` extension.
- The `SessionStore` is now the `WebExtensionDelegate`
- Session life cycle code refactor to establish the `SessionStore` as the single point of communications with the `BrowserStore`.

There are some Glean related updates, mainly initialization changes and lint errors in .md files, could you please @daoshengmu verify that everything is working as expected?

@bluemarvin @MortimerGoro I've have replaced the appcompat aar for the maven version so it stays aligned with the rest of the androidx components. I couldn't find any speech related classes in that module.

**QA**: 
This should affect mainly to the WebExtensions. Things that we should keep an eye on:
- Youtube and Vimeo WebExtensions lifecycle (correct load of the extensions in restored session)
- FxA login: This shouldn't change, we are now using WebChannels but the flow is still the same.
- Would be also worth keeping an eye on sessions state: creation, recreation, shutdown and restore.
